libs/execution_context/ebpf_maps.c

```diff
@@ -1052,128 +1031,137 @@ _Requires_lock_held_(map->lock) static void _uninitialize_lru_entry(
+        break;
     }
-
-    ebpf_list_entry_t* list_entry = map->hot_list.Flink;
-    ebpf_list_remove_entry(&map->hot_list);
-    ebpf_list_append_tail_list(&map->cold_list, list_entry);
-
-    ebpf_list_initialize(&map->hot_list);
-    map->hot_list_size = 0;
-    map->current_generation++;
 }
 
-/**
- * @brief Helper function to update the LRU key history for a given key if the map tracks key history.
- *
- * @param[in,out] map Pointer to the map.
- * @param[in] key Key to update.
- * @param[in] value Optional value for the key. If NULL, the value will be looked up.
- * @param[in] operation Operation to perform on the key.
- * @retval EBPF_SUCCESS The operation was successful.
- * @retval EBPF_KEY_NOT_FOUND The key was not found in the map.
- */
 static ebpf_result_t
-_update_key_history(
-    _Inout_ ebpf_core_map_t* map, _In_ const uint8_t* key, _In_opt_ uint8_t* value, ebpf_lru_key_operation_t operation)
+_create_lru_hash_map(
+    _In_ const ebpf_map_definition_in_memory_t* map_definition,
+    ebpf_handle_t inner_map_handle,
+    _Outptr_ ebpf_core_map_t** map)
 {
-    // Fast path:
-    // If this map doesn't track key history, return success.
-    if (!(ebpf_map_metadata_tables[map->ebpf_map_definition.type].key_history)) {
-        return EBPF_SUCCESS;
-    }
+    ebpf_result_t retval = EBPF_SUCCESS;
+    ebpf_core_lru_map_t* lru_map = NULL;
 
-    // If value was not provided, attempt to find it.
-    if (value == NULL) {
-        ebpf_result_t find_result = ebpf_hash_table_find((ebpf_hash_table_t*)map->data, key, &value);
-        if (find_result != EBPF_SUCCESS) {
-            return find_result;
-        }
-    }
+    *map = NULL;
 
-    // If the value is not found, return success as it has already been deleted.
+    EBPF_LOG_ENTRY();
 
-    // If the value is found, update the key history.
-    if (value) {
-        ebpf_lru_entry_t* entry = (ebpf_lru_entry_t*)_get_value_supplement(map, value);
-        ebpf_core_lru_map_t* lru_map = EBPF_FROM_FIELD(ebpf_core_lru_map_t, core_map, map);
+    if (inner_map_handle != ebpf_handle_invalid) {
+        retval = EBPF_INVALID_ARGUMENT;
+        goto Exit;
+    }
 
-        ebpf_lru_key_state_t key_state = _get_key_state(lru_map, entry);
+    size_t lru_entry_size;
+    retval = ebpf_safe_size_t_add(EBPF_OFFSET_OF(ebpf_lru_entry_t, key), map_definition->key_size, &lru_entry_size);
+    if (retval != EBPF_SUCCESS) {
+        goto Exit;
+    }
 
-        // Fast path:
-        // If the key is already in the hot list, no need to update the key history.
-        if (key_state == EBPF_LRU_KEY_HOT && operation == EBPF_LRU_KEY_OPERATION_UPDATE) {
-            // Key is already in the hot list. No need to update the key history.
-            return EBPF_SUCCESS;
-        }
+    // Align the supplemental value to 8 byte boundary.
+    // Pad value_size to next 8 byte boundary and subtract the value_size to get the padding.
+    size_t supplemental_value_size;
```

Comment:
It seems odd to me that sometimes (1080, 1082, etc.) the term is "supplemental value" and other times (1042, etc.) it's "value supplement".  Assuming they're the same thing, pick one and be consistent.

Reply:
Agree. No idea why I had it backwards there. It should be "supplemental value" throughout now, assuming I didn't miss any.